OutOfSchool/OutOfSchool.WebApi/Services/OrganizationService.cs

```diff
@@ -0,0 +1,134 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using AutoMapper;
+using OutOfSchool.Services.Models;
+using OutOfSchool.Services.Repository;
+using OutOfSchool.WebApi.Mapping.Extensions;
+using OutOfSchool.WebApi.Models;
+
+namespace OutOfSchool.WebApi.Services
+{
+    /// <summary>
+    ///  Service with business logic for Organization model.
+    /// </summary>
+    public class OrganizationService : IOrganizationService
+    {
+        private IOrganizationRepository _repository;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="OrganizationService"/> class.
+        /// </summary>
+        /// <param name="entityRepository">Repository for some entity.</param>
+        public OrganizationService(IOrganizationRepository entityRepository)
+        {
+            _repository = entityRepository;
+        }
+
+        /// <inheritdoc/>
+        public async Task<OrganizationDTO> Create(OrganizationDTO organizationDto)
+        {
+            if (organizationDto == null)
+            {
+                throw new ArgumentNullException(nameof(organizationDto), "Organization was null.");
+            }
+
+            var organization = organizationDto.ToDomain();
+
+            if (!_repository.IsUnique(organization))
+            {
+                throw new ArgumentException(nameof(organization), "There is already an organizationDto with such data");
+            }
+
+            var newOrganization = await _repository.Create(organization).ConfigureAwait(false);
+
+            return newOrganization.ToModel();
+        }
+
+        /// <inheritdoc/>
+        public async Task<IEnumerable<OrganizationDTO>> GetAll()
+        {
+             var organizations = await _repository.GetAll().ConfigureAwait(false);
+            
+             return organizations.Select(organization => organization.ToModel());
+        }
+
+        /// <inheritdoc/>
+        public async Task<OrganizationDTO> GetById(long id)
+        {
+            var organization = await _repository.GetById(id).ConfigureAwait(false);
+            if (organization == null)
+            {
+                throw new ArgumentException("Incorrect Id!", nameof(id));
+            }
+
+            return organization.ToModel();
+        }
+
+        /// <inheritdoc/>
+        public async Task<OrganizationDTO> Update(OrganizationDTO organizationDto)
```

Comment:
organizationDto is quite long
this variable is used many times 
could we name it just 'dto'?
I suppose it should not be confusing but spares a lot of typing and reading of code in this method